qiskit/transpiler/synthesis/permrowcol.py

```diff
@@ -198,3 +195,42 @@ def _eliminate_row(self, parity_mat: np.ndarray, root: int, terminals: np.ndarra
             parity_mat[edge[0], :] = (parity_mat[edge[0], :] + parity_mat[edge[1], :]) % 2
 
         return C
+
+    def _matrix_edit(
+        self, parity_mat: np.ndarray, chosen_column: int, chosen_row: int, circuit: QuantumCircuit
+    ) -> QuantumCircuit:
+        """Checks if the sum of the chosen row is bigger than one. If the sum is bigger than one, the function performs row elimination.
+
+        Args:
+            parity_mat (np.ndarray): parity matrix representing a circuit
+            chosen_column (int): index of the column to be eliminated
+            chosen_row (int): index of the row to be eliminated
+            circuit (QuantumCircuit): quantum circuit to be synthesized
+
+
+        Returns:
+            QuantumCircuit: quantum circuit with chosen row eliminated.
+
+        """
+
+        if sum(parity_mat[chosen_row]) > 1:
```

Comment:
We could move this if statement to perm_row_col to make this method have one clear responsibility